src/core/server/services/tenant/tenant.ts

```diff
@@ -298,8 +305,42 @@ export async function createEmailDomain(
   redis: Redis,
   cache: TenantCache,
   tenant: Tenant,
+  viewer: Pick<User, "id"> | undefined,
   input: CreateEmailDomainInput
 ) {
+  // TODO: This is tech debt, to be removed when
+  // MODERATOR is split into ORG_MODERATOR + SITE_MODERATOR
+  if (!viewer) {
+    throw new UserForbiddenError(
+      "Must be authenticated to create email domain ban",
+      "emailDomain",
+      "create"
+    );
+  }
+
+  const fullViewer = await retrieveUser(mongo, tenant.id, viewer.id);
+  if (!fullViewer) {
+    throw new UserNotFoundError("Viewer not found");
+  }
+
+  const tenantSites = await retrieveTenantSites(mongo, tenant.id);
+  const isAdmin = fullViewer.role === GQLUSER_ROLE.ADMIN;
+  const multiSiteEnabled = tenantSites.length > 1;
+  const modOnSingleSite = isModerator(fullViewer) && !multiSiteEnabled;
+  const orgModOnMultiSite = isOrgModerator(fullViewer) && multiSiteEnabled;
+
+  const allowed =
+    fullViewer.tenantID === tenant.id &&
+    (isAdmin || modOnSingleSite || orgModOnMultiSite);
```

Comment:
Since this is allowing the moderator to be allowed whether they're single site or multi-site, does it matter if they're single or multi-site? Could we do away with retrieving whether the tenant has just one or more sites and simple check if the user's role is a moderator or higher?

Though, I could be wrong, please inform me if I'm missing some edge case here... as always, the single vs multi-site moderators is complicated.

Reply:
If we did that it wouldn't catch the case in which their role is Moderator, but they are a site mod on a multi-site tenant, in which case we don't want them to be able to create email domain bans (according to the ticket). So we want to allow "Org mods on multi site tenants" or "any mod on single site tenants", but not "any mod on multi site tenants".

This is one of the many places where our logic is muddied by Org Mods and Site Mods not being distinct roles. If that were the case, all mods on single site tenants would have the Org Mod role, and we could just check for Org Mod role and ignore whether or not the tenant has multi site enabled.

Reply:
Oh! I'm sorry, I saw `isModerator` and `isOrgModerator` and thought they were the same when I was reviewing. Looking at it fresh right now, I see they are different and this makes sense!